File: rb/features.rb
# Imgflip SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ImgflipFeatures
  def self.make_feature(name)
    case name
    when "base"
      ImgflipBaseFeature.new
    when "ratelimit"
      ImgflipRatelimitFeature.new
    when "retry"
      ImgflipRetryFeature.new
    when "test"
      ImgflipTestFeature.new
    when "timeout"
      ImgflipTimeoutFeature.new
    else
      ImgflipBaseFeature.new
    end
  end
end

File: py/features.py
# Imgflip SDK feature factory

from feature.base_feature import ImgflipBaseFeature
from feature.test_feature import ImgflipTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ImgflipBaseFeature(),
        "test": lambda: ImgflipTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
